lib/api/core/validation/index.js

```diff
@@ -1,690 +1,680 @@
-        if (fieldErrors.length === 0) {
-          // We still want to trigger an error, even if no message is provided
-          manageErrorMessage(field.path, errorMessages, 'An error has occurred during validation.', verbose);
+        if (field.multivalued.hasOwnProperty('maxCount') && documentSubset[fieldName].length > field.multivalued.maxCount) {
+          manageErrorMessage(field.path, errorMessages, `Too many elements. Maximum count is set to ${field.multivalued.maxCount}.`, verbose);
+          return false;
         }
-        else {
-          fieldErrors.forEach(message => manageErrorMessage(field.path, errorMessages, message, verbose));
+
+        fieldValues = documentSubset[fieldName];
+      }
+      else {
+        if (Array.isArray(documentSubset[fieldName])) {
+          manageErrorMessage(field.path, errorMessages, 'The field is not a multivalued field; Multiple values provided.', verbose);
+          return false;
         }
 
-        return false;
+        fieldValues = [documentSubset[fieldName]];
       }
 
-      if (this.types[field.type].allowChildren && field.children) {
-        try {
-          if (!this.recurseFieldValidation(fieldValues[i], field.children, nestedStrictness, errorMessages, verbose)) {
-            result = false;
-          }
-        }
-        catch (error) {
-          if (error.message === 'strictness') {
-            manageErrorMessage(field.path, errorMessages, 'The field is set to "strict"; cannot add unspecified subField.', verbose);
-          }
-          else if (verbose) {
-            manageErrorMessage(field.path, errorMessages, error.message, verbose);
+      if (this.types[field.type].allowChildren) {
+        nestedStrictness = this.types[field.type].getStrictness(field.typeOptions, strictness);
+      }
+
+      for (let i = 0; i < fieldValues.length; i++) {
+        const fieldErrors = [];
+
+        if (!this.types[field.type].validate(field.typeOptions, fieldValues[i], fieldErrors)) {
+          if (fieldErrors.length === 0) {
+            // We still want to trigger an error, even if no message is provided
+            manageErrorMessage(field.path, errorMessages, 'An error has occurred during validation.', verbose);
           }
           else {
-            throw error;
+            fieldErrors.forEach(message => manageErrorMessage(field.path, errorMessages, message, verbose));
           }
 
-          result = false;
+          return false;
+        }
+
+        if (this.types[field.type].allowChildren && field.children) {
+          try {
+            if (!this.recurseFieldValidation(fieldValues[i], field.children, nestedStrictness, errorMessages, verbose)) {
+              result = false;
+            }
+          }
+          catch (error) {
+            if (error.message === 'strictness') {
+              manageErrorMessage(field.path, errorMessages, 'The field is set to "strict"; cannot add unspecified subField.', verbose);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 135 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)